File: app/src/main/java/org/apache/fineract/data/services/LoanService.java
package org.apache.fineract.data.services;

import org.apache.fineract.data.models.loan.LoanAccount;
import org.apache.fineract.data.models.loan.LoanAccountPage;
import org.apache.fineract.data.models.product.ProductPage;
import org.apache.fineract.data.remote.EndPoints;

import io.reactivex.Completable;
import io.reactivex.Observable;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * @author Rajan Maurya
 * On 07/07/17.
 */
public interface LoanService {

    @GET(EndPoints.API_PORTFOLIO_PATH + "/individuallending/customers/{customeridentifier}/cases")
    Observable<LoanAccountPage> fetchCustomerLoanAccounts(
            @Path("customeridentifier") String customerIdentifier,
            @Query("pageIndex") Integer pageIndex,
            @Query("size") Integer size);

    @GET(EndPoints.API_PORTFOLIO_PATH + "/products/{productidentifier}/cases/{caseidentifier}")
    Observable<LoanAccount> fetchCustomerLoanDetails(
            @Path("productidentifier") String productIdentifier,
            @Path("caseidentifier") String caseIdentifier);

    @GET(EndPoints.API_PORTFOLIO_PATH + "/products/")
    Observable<ProductPage> getProducts(
            @Query("pageIndex") Integer pageIndex,
            @Query("size") Integer size,
            @Query("includeDisabled") Boolean includeDisabled);

    @POST(EndPoints.API_PORTFOLIO_PATH + "/products/{productidentifier}/cases/")
    Completable createLoan(
            @Path("productidentifier") String productidentifier,
            @Body LoanAccount loanAccount);

    @PUT(EndPoints.API_PORTFOLIO_PATH + "/products/{productidentifier}/cases/{caseidentifier}")
    Completable updateLoan(
            @Path("productidentifier") String productIdentifier,
            @Path("caseidentifier") String caseIdentifier,
            @Body LoanAccount loanAccount);
}

File: app/src/main/java/org/apache/fineract/data/services/LoanService.kt
package org.apache.fineract.data.services

import io.reactivex.Completable
import io.reactivex.Observable
import org.apache.fineract.data.models.loan.LoanAccount
import org.apache.fineract.data.models.loan.LoanAccountPage
import org.apache.fineract.data.models.product.ProductPage
import org.apache.fineract.data.remote.EndPoints
import retrofit2.http.*

/**
 * Created by Ahmad Jawid Muhammadi on 2/6/20
 */
interface LoanService {

    @GET(EndPoints.API_PORTFOLIO_PATH + "/individuallending/customers/{customeridentifier}/cases")
    fun fetchCustomerLoanAccounts(
            @Path("customeridentifier") customerIdentifier: String?,
            @Query("pageIndex") pageIndex: Int?,
            @Query("size") size: Int?): Observable<LoanAccountPage?>?

    @GET(EndPoints.API_PORTFOLIO_PATH + "/products/{productidentifier}/cases/{caseidentifier}")
    fun fetchCustomerLoanDetails(
            @Path("productidentifier") productIdentifier: String?,
            @Path("caseidentifier") caseIdentifier: String?): Observable<LoanAccount?>?

    @GET(EndPoints.API_PORTFOLIO_PATH + "/products/")
    fun getProducts(
            @Query("pageIndex") pageIndex: Int?,
            @Query("size") size: Int?,
            @Query("includeDisabled") includeDisabled: Boolean?): Observable<ProductPage?>?

    @POST(EndPoints.API_PORTFOLIO_PATH + "/products/{productidentifier}/cases/")
    fun createLoan(
            @Path("productidentifier") productidentifier: String?,
            @Body loanAccount: LoanAccount?): Completable?
}